### ISSUE TITLE
XS snapshot not found during kernel reload

### ISSUE THREAD

**Issue by warner** [Agoric/agoric-sdk#3429]
## Describe the bug

When restarting a local kernel, one of the vats was unable to reload from snapshot because it couldn't find the uncompressed snapshot file:

```
10:19AM INF Timed out dur=3000 height=223 module=consensus round=0 step=3
2021-06-29T17:19:03.301Z SwingSet: kernel: v15 online: xs-worker-no-gc static name=timer transcript entries replayed: 41
cannot read snapshot /home/warner/stuff/agoric/tlg2/_agstate/agoric-servers/local-chain-26657/data/ag-cosmos-chain-state/xs-snapshots/39c60159056c93c7ac82ca36cbf8d8bd07ee972491a0f064c594cd58b1beeee1-load-Ltrw9Y.xss: No such file or directory
Cannot initialize Controller Error: Cannot write messages to v12:vatAdmin: This socket has been ended by the other party
```

When I looked afterwards, the corresponding compressed file does seem to exist in that `xs-snapshots/` directory:

```
warner@turing:~/stuff/agoric/tlg2$ ls -l _agstate/agoric-servers/local-chain-26657/data/ag-cosmos-chain-state/xs-snapshots/39c6015905*
-rw-r--r-- 1 warner warner 296399 Jun 29 09:59 _agstate/agoric-servers/local-chain-26657/data/ag-cosmos-chain-state/xs-snapshots/39c60159056c93c7ac82ca36cbf8d8bd07ee972491a0f064c594cd58b1beeee1.gz
```

It launched normally when I restarted the chain process a second time.

The same thing happened to my solo node, and again a restart allows it to proceed.

I'm suspecting something involving the tempfile used for the decompressed version. Maybe we aren't waiting for the decompression to complete before giving it to `xsnap`?

**Comment by dckc** [comment]
> I'm suspecting something involving the tempfile used for the decompressed version. Maybe we aren't waiting for the decompression to complete before giving it to `xsnap`?

right... the `.evaluate('null')` ensured that the subprocess was done loading the snapshot before the snapstore removed it. You saw these symptoms when you took out the `.evaluate('null')`. #3433 replaces `.evaluate('null')` with `.isRead()` which does the synchronization without perturbing the XS heap.

I'm inclined to close this, since the symptoms don't happen with the `master` branch, but I suppose it can wait 'till #3433 lands.

**Comment by warner** [comment]
yeah, let's mark #3433 as closing this one

**Comment by dckc** [comment]
fixed in #3433 / 6e4cf0c
